app/assets/javascripts/discourse/app/components/topic-list/topic-list-header.gjs

```diff
@@ -1,116 +1,176 @@
-    />
+  @cached
+  get columns() {
+    const self = this;
+    const context = {
+      get category() {
+        return self.topicTrackingState.get("filterCategory");
+      },
 
-    {{#if @showLikes}}
-      <TopicListHeaderColumn
-        @sortable={{@sortable}}
-        @number="true"
-        @order="likes"
-        @activeOrder={{@order}}
-        @changeSort={{@changeSort}}
-        @ascending={{@ascending}}
-        @name="likes"
-      />
-    {{/if}}
+      get filter() {
+        return self.topicTrackingState.get("filter");
+      },
+    };
 
-    {{#if @showOpLikes}}
-      <TopicListHeaderColumn
-        @sortable={{@sortable}}
-        @number="true"
-        @order="op_likes"
-        @activeOrder={{@order}}
-        @changeSort={{@changeSort}}
-        @ascending={{@ascending}}
-        @name="likes"
-      />
-    {{/if}}
+    return applyValueTransformer(
+      "topic-list-header-columns",
+      createColumns(),
+      context
+    );
+  }
 
-    <TopicListHeaderColumn
-      @sortable={{@sortable}}
-      @number="true"
-      @order="views"
-      @activeOrder={{@order}}
-      @changeSort={{@changeSort}}
-      @ascending={{@ascending}}
-      @name="views"
-    />
-
-    <TopicListHeaderColumn
-      @sortable={{@sortable}}
-      @number="true"
-      @order="activity"
-      @activeOrder={{@order}}
-      @changeSort={{@changeSort}}
-      @ascending={{@ascending}}
-      @name="activity"
-    />
-
-    <PluginOutlet @name="topic-list-header-after" />
-  </tr>
-</template>;
-
-export default TopicListHeader;
+  <template>
+    <tr>
+      {{#each (this.columns.resolve) as |entry|}}
+        {{#if entry.value}}
+          <entry.value
+            @sortable={{@sortable}}
+            @activeOrder={{@order}}
+            @changeSort={{@changeSort}}
+            @ascending={{@ascending}}
+            @category={{@category}}
+            @name={{@listTitle}}
+            @bulkSelectEnabled={{@bulkSelectEnabled}}
+            @showBulkToggle={{@toggleInTitle}}
+            @canBulkSelect={{@canBulkSelect}}
+            @canDoBulkActions={{@canDoBulkActions}}
+            @showTopicsAndRepliesToggle={{@showTopicsAndRepliesToggle}}
+            @newListSubset={{@newListSubset}}
+            @newRepliesCount={{@newRepliesCount}}
+            @newTopicsCount={{@newTopicsCount}}
+            @bulkSelectHelper={{@bulkSelectHelper}}
+            @changeNewListSubset={{@changeNewListSubset}}
+          />
+        {{else if (eq entry.key "bulk-select")}}
+          {{#if @bulkSelectEnabled}}
+            <th class="bulk-select topic-list-data">
+              {{#if @canBulkSelect}}
+                <button
+                  {{on "click" @bulkSelectHelper.toggleBulkSelect}}
+                  title={{i18n "topics.bulk.toggle"}}
+                  class="btn-flat bulk-select"
+                >
+                  {{icon "list-check"}}
+                </button>
+              {{/if}}
+            </th>
+          {{/if}}
```

Comment:
What's the reason for making these a whole sequence of `if()` branches? How about making each of them a template-only-component, and then including the components as the values in the DAG?

Then, the core table cells will work the same as plugin/theme-contributed table cells.